diagnostics-extension/src/telemetry.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Classes related to telemetry
+ */
+
+import {
+  BacklightInfo,
+  BatteryInfo,
+  BlockDeviceInfo,
+  BluetoothInfo,
+  CpuInfo,
+  FanInfo,
+  MemoryInfo,
+  StatefulPartitionInfo,
+  TimezoneInfo,
+  VpdInfo,
+} from './dpsl.types';
+import * as fakeData from './fake_telemetry.data';
+
+/**
+ * Abstract class reprensenting the interface of
+ * service to fetch system telemetry data
+ */
+abstract class TelemetryService {
+  abstract getBacklightInfo(): Promise<BacklightInfo>;
+  abstract getBatteryInfo(): Promise<BatteryInfo>;
+  abstract getBluetoothInfo(): Promise<BluetoothInfo>;
+  abstract getCachedVpdInfo(): Promise<VpdInfo>;
+  abstract getCpuInfo(): Promise<CpuInfo>;
+  abstract getFanInfo(): Promise<FanInfo>;
+  abstract getMemoryInfo(): Promise<MemoryInfo>;
+  abstract getNonRemovableBlockDevicesInfo(): Promise<BlockDeviceInfo>;
+  abstract getStatefulPartitionInfo(): Promise<StatefulPartitionInfo>;
+  abstract getTimezoneInfo(): Promise<TimezoneInfo>;
+}
+
+/**
+ * Class to fetch fake system telemetry data
```

Comment:
nit:
* Fake implementation of TelemetryService. Used in unittests.
* Returns fake data from fake_telemetrt.data.